app.js

```diff
@@ -1,25 +1,57 @@
-const express = require('express')
-const logger = require('morgan')
-const cors = require('cors')
-
-const contactsRouter = require('./routes/api/contacts')
-
-const app = express()
-
-const formatsLogger = app.get('env') === 'development' ? 'dev' : 'short'
-
-app.use(logger(formatsLogger))
-app.use(cors())
-app.use(express.json())
-
-app.use('/api/contacts', contactsRouter)
+const express = require("express");
+const logger = require("morgan");
+const cors = require("cors");
+const Joi = require("joi");
+
+const contactsRouter = require("./routes/api/contacts");
+
+const app = express();
+
+const formatsLogger = app.get("env") === "development" ? "dev" : "short";
+
+app.use(logger(formatsLogger));
+app.use(cors());
+app.use(express.json());
+
+app.use("/api/contacts", contactsRouter);
+
+app.get("/", (req, res) => {
+  res.send("Welcome to the Contact API");
+});
+
+const validateContact = (contact) => {
+  const schema = Joi.object({
+    name: Joi.string().required(),
+    email: Joi.string().email().required(),
+    phone: Joi.string().required(),
+  });
+  return schema.validate(contact);
+};
+
+app.post("/api/contacts", (req, res) => {
+  const { error } = validateContact(req.body);
+  if (error) {
+    return res.status(400).json({ message: error.details[0].message });
+  }
+});
+
+app.put("/api/contacts/:id", (req, res) => {
```

Comment:
if you have `app.use("/api/contacts", contactsRouter);` above why do you have this handler?